### PR TITLE
Introduces TimeLayerSliderPanel

### PR DESCRIPTION
**Pull request by KaiVolland**
This is almost a copy of #436 as the project structure has changed and creating a new PR was easier then rebasing.

Credits go to @ahennr .

closes #436

### CODE REVIEW

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    // this.sunRiseLayer = new SunRiseLayer({
+    //   map: this.props.map
+    // });
+
+    this.state = {
+      value: moment("2015-01-01"),
+      forceUpdate: 1,
+      playbackSpeed: '1',
+      autoPlayActive: false,
+      startDate: moment("2006-01-01"),
```
Comment (annarieger) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: Do we always want to have the same start and current date values regardless of the layer? These values should besser depend on real data I think..

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            icon="calendar-o"
+          />
+        </Popover>
+        {
+          resetVisible ?
+            <SimpleButton
+              type="primary"
+              icon="refresh"
+              onClick={this.setSliderToNow}
+              tooltip={'Auf aktuelle Zeit zentrieren'}
```
Comment (annarieger) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: i18ns

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+          <Option value="2">2x</Option>
+          <Option value="5">5x</Option>
+          <Option value="10">10x</Option>
+          <Option value="100">100x</Option>
+          <Option value="300">300x</Option>
+          <Option value="hours">hour</Option>
+          <Option value="days">day</Option>
+          <Option value="weeks">week</Option>
+          <Option value="months">month</Option>
+          <Option value="years">year</Option>
```
Comment (annarieger) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: i18ns

File: src/config/config.js [hunk 4]
```diff
@@ -8,10 +8,7 @@ let localePath =  basePath + 'resources/i18n/{{lng}}.json';
 const appMode = typeof(APP_MODE) != "undefined" ? APP_MODE : undefined;
 const nodeEnv = typeof(process.env.NODE_ENV) != "undefined" ? process.env.NODE_ENV : undefined;
 
-
-if (!nodeEnv || nodeEnv.indexOf('development') > -1) {
-  localePath =  basePath + 'resources/i18n/{{lng}}.json';
-} else if (nodeEnv.indexOf('production') > -1) {
+if (nodeEnv.indexOf('production') > -1) {
```
Comment (weskamm) [inline src/config/config.js]: nodeEnv may still be undefined here, so this can fail

Comment (KaiVolland) [reply src/config/config.js]: Fixed it in a force push.

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.less [hunk 5]
```diff
@@ -0,0 +1,47 @@
+.time-layer-slider {
+  position: absolute;
+  bottom: 35px;
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.less]: Absolute positing in a component is horrible anyways. :laughing:
We should remove it when moving the components over to [react-geo](https://github.com/terrestris/react-geo).

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+  /**
+  * makes sure that the appended time parameter in GetMap calls
+  * is rounded to full hours to receive a valid response
+  */
+  wmsTimeHandler = (value?: any) => {
+    this._wmsTimeLayers.forEach(config => {
+      if (config.layer && config.layer.get('type') === 'WMSTime') {
+        const params = config.layer.getSource().getParams();
+        let time;
+        if (Array.isArray(value)) {
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: It is perfectly valid to have a range for a time layer.

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+          newValue = value.clone().add(1, valueToAdd as moment.DurationInputArg2);
+        }
+
+        this.timeSliderCustomHandler(newValue);
+        this.wmsTimeHandler(newValue);
+        // value is handled in timeSliderCustomHandler
+        this.setState({
+          autoPlayActive: true
+        });
+      }, 1000, this);
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: Hardcoded `1000`

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+    this.state = {
+      value: props.value,
+      forceUpdate: 1,
+      playbackSpeed: '1',
+      autoPlayActive: false,
+      startDate: props.startDate,
+      endDate: props.endDate
+    };
+
+    this._interval = 1000;
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: Most probably unneeded, as this is only used for the `setInterval`, right?

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 9]
```diff
@@ -0,0 +1,371 @@
...
+    this.wrapTimeSlider();
+  }
+
+  /**
+  * Wraps the TimeSlider component in timeLayerAware.
+  */
+  wrapTimeSlider = () => {
+    this._wmsTimeLayers = [];
+    this.props.timeAwareLayers!.forEach((l: any) => {
+      if (l.get('type') === 'WMSTime') {
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: We should probably have a layer type lookup with constants

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+        if (Array.isArray(value)) {
+          time = value[0];
+        } else {
+          time = value;
+        }
+        if (!moment.isMoment(time)) {
+          time = moment(time);
+        }
+        time.set('minute', 0);
+        time.set('second', 0);
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: Why? This should be configurable

File: src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+    marks[mid.toISOString()] = {
+      label: mid.format(dateFormat)
+    };
+
+    return (
+      <div className="time-layer-slider">
+
+        <Popover
+          placement="topRight"
+          title="Zeitleisten Bereich"
```
Comment (marcjansen) [inline src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx]: i18n

**Pull request merged**